base/shell.jl

```diff
@@ -12,21 +12,20 @@ function shell_parse(str::AbstractString, interpolate::Bool=true;
                      special::AbstractString="")
     s = lstrip(str)
     # strips the end but respects the space when the string ends with "\\ "
-    r = RevString(s)
-    i = start(r)
-    c_old = nothing
-    while !done(r,i)
-        c, j = next(r,i)
+    i = endof(s)
+    c_old = '\0' # initialized to a null byte for type stability
+    while i > 0
+        c = s[i]
         if c == '\\' && c_old == ' '
-            i -= 1
+            i += 1
             break
-        elseif !(c in _default_delims)
+        elseif !in(c, _default_delims)
```

Comment:
not sure why this changed?  The old way seemed clearer to me.  Or you could do `c ∉ _default_delims`

Reply:
Huh, I don't know why I changed that either. Will revert.